Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java

```diff
@@ -74,391 +76,348 @@
-				}catch(QuadrigaStorageException e){
-					logger.error("Errors in adding items",e);
-					response.setStatus(500);
-					response.setContentType(accept);
-					String errorMsg = errorMessageRest.getErrorMsg("Failed to add due to DB Error",request);
-					return errorMsg;
-				}
-
-			}
-			
-			response.setStatus(200);
-			response.setContentType(accept);
-			return "Success";
-		}
-	}
-	
+    private static final Logger logger = LoggerFactory.getLogger(DictionaryRestController.class);
+
+    @Autowired
+    private IWorkspaceDictionaryManager workspaceDictionaryManager;
+
+    @Autowired
+    private IWSSecurityChecker checkWSSecurity;
+
+    @Autowired
+    private IUserManager usermanager;
+
+    @Autowired
+    private IRestMessage errorMessageRest;
+
+    @Autowired
+    private IDictionaryManager dictionaryManager;
+
+    @Autowired
+    private IDictionaryFactory dictionaryFactory;
+
+    @Autowired
+    private IRestVelocityFactory restVelocityFactory;
+
+    @Autowired
+    @Qualifier("updateFromWordPowerURLPath")
+    private String updateFromWordPowerURLPath;
+
+    @Autowired
+    @Qualifier("wordPowerURL")
+    private String wordPowerURL;
+
+    /**
+     * Rest interface for the List Dictionary for the userId http://<<URL>:
+     * <PORT>>/quadriga/rest/dictionaries
+     * http://localhost:8080/quadriga/rest/dictionaries
+     * 
+     * @author Lohith Dwaraka
+     * @param userId
+     * @param model
+     * @return
+     * @throws Exception
+     */
+    @RequestMapping(value = "rest/dictionaries", method = RequestMethod.GET, produces = "application/xml")
+    public ResponseEntity<String> listDictionaries(ModelMap model, Principal principal, HttpServletRequest req)
+            throws Exception {
+        UserDetails user = (UserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        List<IDictionary> dictionaryList = null;
+        VelocityEngine engine = restVelocityFactory.getVelocityEngine(req);
+
+        Template template = null;
+
+        try {
+            engine.init();
+            dictionaryList = dictionaryManager.getDictionariesList(user.getUsername());
+            template = engine.getTemplate("velocitytemplates/dictionarylist.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("list", dictionaryList);
+            StringWriter writer = new StringWriter();
+            template.merge(context, writer);
+            return new ResponseEntity<String>(writer.toString(), HttpStatus.OK);
+        } catch (ResourceNotFoundException e) {
+            logger.error("Exception:", e);
+            throw new RestException(404);
+        } catch (ParseErrorException e) {
+            logger.error("Exception:", e);
+            throw new RestException(404);
+        } catch (MethodInvocationException e) {
+            logger.error("Exception:", e);
+            throw new RestException(403);

```

Comment:
wrap the original exception in the new one and remove the logger (same in the ohter classes).